yarn-project/foundation/src/fifo/memory_fifo.ts

```diff
@@ -39,6 +39,10 @@ export class MemoryFifo<T> {
       return Promise.resolve(null);
     }
 
+    if (timeout === 0) {
```

Comment:
As it stands, returning `null` means it's flushing I think. Should it perhaps raise a timeout error, or maybe another type of error.

Reply:
You're absolutely right, I totally missed that. I think `TimeoutError` makes sense since it technically times out immediately (next call frame)